file_metadata/utilities.py

```diff
@@ -1,35 +1,132 @@
 # -*- coding: utf-8 -*-
 
+from __future__ import (division, absolute_import, unicode_literals,
+                        print_function)
+
 import os
-import sys
 import tempfile
+from shutil import copyfileobj
+
+try:
+    from urllib.request import urlopen  # Python 3
+except ImportError:
+    from urllib2 import urlopen  # Python 2
 
 from contextlib import contextmanager
 
 
 PROJECT_PATH = os.path.abspath(os.path.dirname(__file__))
 
 
-if sys.version_info < (3,):
-    import codecs
+def download(url, filename, overwrite=False):
+    """
+    Download the given URL to the given filename. If the file exists,
+    it won't be downloaded unless asked to overwrite. Both, text data
+    like html, txt, etc. or binary data like images, audio, etc. are
+    acceptable.
 
-    def u(x):
-        return codecs.unicode_escape_decode(x)[0]
-else:
-    def u(x):
-        return x
+    :param url:       A URL to download.
+    :param filename:  The file to store the downloaded file to.
+    :param overwrite: Set to True if the file should be downloaded even if it
+                      already exists.
+    """
+    if not os.path.exists(filename) or overwrite:
+        response = urlopen(url)
+        with open(filename, 'wb') as out_file:
+            copyfileobj(response, out_file)
 
 
 @contextmanager
-def make_temp(suffix="", prefix="tmp", dir=None):
+def make_temp(suffix="", prefix="tmp", directory=None):
     """
```

Comment:
The code does not comply to PEP8.

*PEP8Bear, severity NORMAL, section `autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/file_metadata/utilities.py
+++ b/file_metadata/utilities.py
@@ -123,6 +123,7 @@
     Computing the answer ...
     42
     """
+
     def __init__(self, wrapped_function):
         self.wrapped_function = wrapped_function
 
```